cmd/client/resources/object.go

```diff
@@ -272,61 +282,90 @@ func unmarshalPrintNamespaceMetaSpec(body []byte, filter func(*supervisor.MetaSp
 	return nil
 }
 
-func unmarshalMetaSpec(body []byte, listBody bool) ([]*supervisor.MetaSpec, error) {
+func unmarshalMetaSpec(body []byte, listBody bool) ([]*MetaSpec, error) {
 	if listBody {
-		metas := []*supervisor.MetaSpec{}
+		metas := []*MetaSpec{}
 		err := codectool.Unmarshal(body, &metas)
 		return metas, err
 	}
-	meta := &supervisor.MetaSpec{}
+	meta := &MetaSpec{}
 	err := codectool.Unmarshal(body, meta)
-	return []*supervisor.MetaSpec{meta}, err
+	return []*MetaSpec{meta}, err
 }
 
-func unmarshalNamespaceMetaSpec(body []byte) (map[string][]*supervisor.MetaSpec, error) {
-	res := map[string][]*supervisor.MetaSpec{}
+func unmarshalNamespaceMetaSpec(body []byte) (map[string][]*MetaSpec, error) {
+	res := map[string][]*MetaSpec{}
 	err := codectool.Unmarshal(body, &res)
 	return res, err
 }
 
-func getAgeFromMetaSpec(meta *supervisor.MetaSpec) string {
+func getAgeFromMetaSpec(meta *MetaSpec) string {
 	createdAt, err := time.Parse(time.RFC3339, meta.CreatedAt)
 	if err != nil {
 		return "unknown"
 	}
 	return general.DurationMostSignificantUnit(time.Since(createdAt))
 }
 
-func printNamespaceMetaSpec(metas map[string][]*supervisor.MetaSpec) {
+func printNamespaceMetaSpec(metas map[string][]*MetaSpec) {
 	// Output:
+	// NAME KIND NAMESPACE PORT HTTPS AGE
+	// ...
+	//
 	// NAME KIND NAMESPACE AGE
 	// ...
+	tableHTTPServer := [][]string{}
+	tableHTTPServer = append(tableHTTPServer, []string{"NAME", "KIND", "NAMESPACE", "PORT", "HTTPS", "AGE"})
 	table := [][]string{}
 	table = append(table, []string{"NAME", "KIND", "NAMESPACE", "AGE"})
+
 	defaults := metas[DefaultNamespace]
 	for _, meta := range defaults {
-		table = append(table, []string{meta.Name, meta.Kind, DefaultNamespace, getAgeFromMetaSpec(meta)})
+		if meta.Kind == "HTTPServer" {
+			tableHTTPServer = append(tableHTTPServer, []string{meta.Name, meta.Kind, DefaultNamespace, strconv.Itoa(meta.Port), strconv.FormatBool(meta.HTTPS), getAgeFromMetaSpec(meta)})
+		} else {
+			table = append(table, []string{meta.Name, meta.Kind, DefaultNamespace, getAgeFromMetaSpec(meta)})
+		}
 	}
+
 	for namespace, metas := range metas {
 		if namespace == DefaultNamespace {
 			continue
 		}
 		for _, meta := range metas {
-			table = append(table, []string{meta.Name, meta.Kind, namespace, getAgeFromMetaSpec(meta)})
+			if meta.Kind == "HTTPServer" {
```

Comment:
It is common for each object to have its representation, so a well-designed framework is needed to handle the difference.

Reply:
updated, please check